election/tests.py

```diff
@@ -148,25 +151,57 @@ def test_vote_wrong_token(self):
             urls.vote(),
             {
                 "election": self.election.id,
-                "mentions_by_candidate": [0, 0],
+                "grades_by_candidate": [0, 0],
                 # make sure the token is not the good one
                 "token": self.token.id + "#abc"
             }
         )
 
         self.assertEqual(400, response.status_code)
 
-    def test_vote_alreadry_used_token(self):
+    def test_vote_already_used_token(self):
         for _ in range(2):
             response = self.client.post(
                 urls.vote(),
                 {
                     "election": self.election.id,
-                    "mentions_by_candidate": [0, 0],
+                    "grades_by_candidate": [0, 0],
                     "token": self.token.id
                 }
             )
 
 
         self.assertEqual(400, response.status_code)
 
+
+class VoteWithResutsTestCase(TestCase):
+
+     def setUp(self):
+
+        self.election = Election.objects.create(
+            title="Test election",
+            candidates=[
+                "Seb",
+                "Pierre-Louis",
+            ], 
+            num_grades=5,
+            on_invitation_only=True,
+        )
+
+        self.votes = [
+             Vote.objects.create(election=self.election, grades_by_candidate=[1,2]),
+             Vote.objects.create(election=self.election, grades_by_candidate=[3, 2]),
+             Vote.objects.create(election=self.election, grades_by_candidate=[1,3]),
+             Vote.objects.create(election=self.election, grades_by_candidate=[4,1])
+        ]
+
+
```

Comment:
maybe we could add more tests to gain confidence in the algorithm. Did you look Clément's tests https://github.com/oceanBigOne/MajorityJudgment/blob/master/demo/fixture01.php ?

Reply:
Yup his fixture came from mine (https://github.com/MieuxVoter/moje). More tests are a really good idea, but that will be the subject of my next PR ;) let's ship that one first!